arrow_util/src/optimize.rs

```diff
@@ -501,4 +549,29 @@ mod tests {
         ]);
         assert_eq!(array, &expected)
     }
+
+    #[test]
+    fn test_split_batch_for_grpc_response() {
+        // no split
+        let c = UInt32Array::from(vec![1, 2, 3, 4, 5, 6]);
+        let batch = RecordBatch::try_from_iter(vec![("a", Arc::new(c) as ArrayRef)])
+            .expect("cannot create record batch");
+        let split = split_batch_for_grpc_response(batch.clone());
+        assert_eq!(split.len(), 1);
+        assert_eq!(batch, split[0]);
+
+        // split once
+        let n_rows = MAX_GRPC_RESPONSE_BATCH_SIZE + 1;
```

Comment:
I think the test would be clearer if you hard coded the number of rows here -- it is somewhat confusing that a batch size of bytes is used as the number of rows in a test

Reply:
I missed the point that num rows ~ num bytes because it is making a `UInt8Array` 👍  sorry for the confusion

```suggestion
        // Choose n_rows ~ n_bytes by creating an Int8Array
        let n_rows = MAX_GRPC_RESPONSE_BATCH_SIZE + 1;
```